examples/person_search/demo.ipynb

```diff
@@ -0,0 +1,399 @@
+    "            # iterate features from the last 30 frames\n",
+    "            if query_feature is not None:\n",
+    "                dist = np.dot(gallery_features, query_feature)  # cosine similarity distance\n",
+    "                past_ids.append(np.argmax(dist))  # the most similar IDs\n",
+    "                past_dist.append(np.max(dist))  # the most similarity distance\n",
+    "\n",
+    "        ids = np.argmax(np.bincount(past_ids))  # the most matched IDs\n",
+    "        dist = np.mean(past_dist)  # the mean distance over past matching\n",
+    "\n",
+    "        return ids, dist\n",
+    "\n",
+    "\n",
+    "# load pre-trained models for person feature extracting\n",
+    "sys.path.append(\"VQPy/examples/person_search/\")\n",
+    "from models import ReIDPredictor\n",
+    "feature_predictor = ReIDPredictor(cfg=\"MSMT17/bagtricks_R50.yml\")\n",
+    "\n",
+    "# extract the feature of query images\n",
+    "gallery_features = []\n",
+    "for file_name in os.listdir(query_folder):\n",
+    "    # extract features for all images from given directory\n",
+    "    img_path = os.path.join(query_folder, file_name)\n",
+    "    preds = feature_predictor(img_path)\n",
+    "    gallery_features.append(preds)\n",
+    "\n",
+    "gallery_features = np.concatenate(gallery_features, axis=0)\n",
+    "\n",
+    "Person.feature_predictor = feature_predictor\n",
+    "Person.gallery_features = gallery_features"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Query on `Person` retrieval\n",
+    "\n",
+    "The `Person` VObj has the `candidate` property to describe the most similar query object IDs and the corresponding score. We pre-defined a threshold `0.97` for candidate score to filter out the matching person in videos.\n",
+    "\n",
+    "`filter_cons` is:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filter_cons = {\n",
+    "    '__class__': lambda x: x == Person,\n",
+    "    'candidate': lambda x: x[1] >= 0.97,  # similar threshold\n",
+    "}"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For output, we select:\n",
+    "\n",
+    "- tracker id, selected with `track_id`\n",
+    "- candidate id, selected with the `candidate[0]`. Need to be converted to `str` before serializing.\n",
+    "- bounding box, in format of coordinates of top-left and bottom-right corner, selected with `tlbr`. Need to be converted to `str` before serializing.\n",
+    "\n",
+    "`select_cons` is:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "select_cons = {\n",
+    "    'track_id': None,\n",
+    "    'candidate': lambda x: str(x[0]),  # convert IDs to string\n",
```

Comment:
Nit: it seems the name is different from that in `expected_results`, which is "match"

Reply:
Thanks! I have checked and updated the expected results. The obsolete key `match` is changed to `candidate`.